src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -577,14 +577,16 @@ private static CommonCompilerOptions GetCompilationOptions(JsonObject rawObject,
 
         private static string MakeDefaultTargetFrameworkDefine(NuGetFramework targetFramework)
         {
-            var shortName = targetFramework.GetTwoDigitShortFolderName();
+            var candidateName = targetFramework.GetTwoDigitShortFolderName().ToUpperInvariant();
 
-            if (targetFramework.IsPCL)
+            if (targetFramework.IsPCL && string.IsNullOrWhiteSpace(targetFramework.Profile))
             {
                 return null;
             }
-
-            var candidateName = shortName.ToUpperInvariant();
+            else
+            {
+                candidateName = $"{FrameworkConstants.FrameworkIdentifiers.Portable.TrimStart('.')}_{targetFramework.Profile}".ToUpperInvariant();
```

Comment:
I don't see how this is right. Doesn't this happen for all targets? I also think we should test this.